etcddb/etcddb_client.go

```diff
@@ -48,20 +51,40 @@ func NewEtcdClientFromVip(vip *viper.Viper,metaData *blockchain.OrganizationMeta
 	conf, err := GetEtcdClientConf(vip,metaData)
 
 	if err != nil {
-		return
+		return nil,err
 	}
 
 	log.WithField("PaymentChannelStorageClient", fmt.Sprintf("%+v", conf)).Info()
 
-	etcdv3, err := clientv3.New(clientv3.Config{
-		Endpoints:   metaData.GetPaymentStorageEndPoints(),
-		DialTimeout: conf.ConnectionTimeout,
-	})
+	var etcdv3 *clientv3.Client
 
 	if err != nil {
-		return
+		return nil,err
+	}
+
+	if tlsConfig,err := getTlsConfig(metaData); err != nil {
+		return nil,err
+	}else if tlsConfig != nil {//https call, with tls config
+		etcdv3, err = clientv3.New(clientv3.Config{
+			Endpoints:   metaData.GetPaymentStorageEndPoints(),
+			DialTimeout: conf.ConnectionTimeout,
+			TLS:tlsConfig,
+
+		})
+
+	}else {
+		//Regular http call
+		etcdv3, err = clientv3.New(clientv3.Config{
+			Endpoints:   metaData.GetPaymentStorageEndPoints(),
+			DialTimeout: conf.ConnectionTimeout,
+
+		})
+		if err != nil {
+			return nil,err
+		}
 	}
```

Comment:
If there is any issue with the certificates the connection will switch to http but the connection string will be https. Yes, etcd will throw an exception.
Is there any better way to check if it is https then use tls otherwise go with the http. We can directly check from the endpoints whether we are going to use http or https based on that we can have the cert validation and then the connection validation.

Reply:
I can see the checkIfHttps in getTlsConfig, see if we can push the condition outside of the function. where getTlsConfig will be called only if it is https.